### PR TITLE
Update BatchNorm to specify population variance

### PR DESCRIPTION
**Pull request by neginraoof**
Update BatchNorm to specify population variance.

Signed-off-by: neginraoof <neginmr@utexas.edu>

### CODE REVIEW

**Review by askhade — APPROVED** [review]

**Review by jcwchen — APPROVED** [review]
The updated .onnx testdata is good @onnx/sig-operators-approvers please forward it. Thanks.

File: docs/Changelog.md [hunk 1]
```diff
@@ -18648,6 +18648,11 @@ This version of the operator has been available since version 14 of the default
   
   current_mean = ReduceMean(X, axis=all_except_channel_index)
   current_var =  ReduceVar(X, axis=all_except_channel_index)
+  
+  Notice that ReduceVar refers to the population variance, and it equals to
+  sum(sqrd(x_i - x_avg)) / N
+  where N is the population size (this formula does not use sample size N - 1).
```
Comment (jcwchen) [inline docs/Changelog.md:18654]: Don't forget to consider the comment here: https://github.com/onnx/onnx/pull/3406#discussion_r608826460

**Review by gramalingam — APPROVED** [review]

**Comment by SherlockNoMad** [comment]
During mixed precision training, input_mean, input_var and running_mean and running_var are usually kept in fp32, while X and Y can be in fp16.

see https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html#lossscaling 

We need to introduce a different type for input_mean, input_var and running_mean and running_var.

**Review by SherlockNoMad — CHANGES_REQUESTED** [review]
need a new type for input_mean, input_var and running_mean and running_var

**Comment by askhade** [comment]
The corresponding cherry pick change has been already checked in release branch so we should not make any more changes to this PR... any changes should be made in a new PR

**Pull request merged**